pkg/cortex/cortex.go

```diff
@@ -193,6 +199,33 @@ func (c *Config) Validate(log log.Logger) error {
 	return nil
 }
 
+// validateYAMLEmptyNodes ensure that no empty node has been specified in the YAML config file.
+// When an empty node is defined in YAML, the YAML parser sets the whole struct to its zero value
+// and so we loose all default values. It's very difficult to detect this case for the user, so we
+// try to prevent it (on the root level) with this custom validation.
+func (c *Config) validateYAMLEmptyNodes() error {
+	defaults := Config{}
+	flagext.DefaultValues(&defaults)
+
+	defStruct := reflect.ValueOf(defaults)
+	cfgStruct := reflect.ValueOf(*c)
+
+	// We expect all structs are the exact same. This check should never fail.
+	if cfgStruct.NumField() != defStruct.NumField() {
+		return errors.New("unable to validate configuration because of mismatching internal config data structure")
+	}
+
+	for i := 0; i < cfgStruct.NumField(); i++ {
+		// If the struct has been reset due to empty YAML value and the zero struct value
+		// doesn't match the default one, then we should warn the user about the issue.
+		if cfgStruct.Field(i).Kind() == reflect.Struct && cfgStruct.Field(i).IsZero() && !defStruct.Field(i).IsZero() {
+			return fmt.Errorf("the %s configuration in YAML has been specified as an empty YAML node", cfgStruct.Type().Field(i).Name)
```

Comment:
I'm not sure I would understand this error message. Given that, should there be an option to use my (seemingly wrong) config anyway?